From: Ralph Giles <giles@mozilla.com>
Bug 697881 - Fall back to gcc on darwin if there's no gcc-4.2. r=ted
Since Bug 513353, we've defaulted to gcc-4.2 on darwin.
Unfortunately the default was unconditional, and while
XCode 4.1 provided a gcc-4.2 alias, XCode 4.2 does not,
although 'gcc' is llvm-gcc 4.2.1. As such, the configure
script immediately fails.

The patch still prefers gcc-4.2, but falls back to plain
gcc if it's not available.

Note that we try $CC and $CXX from the environment first
so it is still possible to override this default.


diff -ruN mozilla-release-orig/configure.in mozilla-release/configure.in
--- mozilla-release-orig/configure.in	2011-09-02 16:15:20.000000000 -0400
+++ mozilla-release/configure.in	2011-11-01 17:06:31.000000000 -0400
@@ -199,18 +199,21 @@
         x86_64)
             AS=ml64;
             ;;
         esac
     fi
     if test -z "$MIDL"; then MIDL=midl; fi
     ;;
 *-darwin*)
-    if test -z "$CC"; then CC=gcc-4.2; fi
-    if test -z "$CXX"; then CXX=g++-4.2; fi
+    # prefer gcc-4.2 to default cc on older darwin, so
+    # use that specific version if it's available.
+    # Bug 697881 - Will be in Firefox10.
+    MOZ_PATH_PROGS(CC, $CC gcc-4.2 gcc)
+    MOZ_PATH_PROGS(CXX, $CXX g++-4.2 g++)
     ;;
 esac
 fi
 
 COMPILE_ENVIRONMENT=1
 MOZ_ARG_DISABLE_BOOL(compile-environment,
 [  --disable-compile-environment
                           Disable compiler/library checks.],
diff --git a/js/src/configure.in b/js/src/configure.in
index 499154c..806a20b 100644
--- a/js/src/configure.in
+++ b/js/src/configure.in
@@ -168,17 +168,17 @@ dnl ==============================================================
 MOZ_ARG_WITH_STRING(dist-dir,
 [  --with-dist-dir=DIR     Use DIR as 'dist' staging area.  DIR may be
                           relative to the top of SpiderMonkey build tree,
                           or absolute.],
     TOP_DIST=$withval,
     TOP_DIST=dist)
 AC_SUBST(TOP_DIST)
 
-dnl Default to MSVC for win32 and gcc-4.2 for darwin
+dnl Default to MSVC for win32
 dnl ==============================================================
 if test -z "$CROSS_COMPILE"; then
 case "$target" in
 *-mingw*)
     if test -z "$CC"; then CC=cl; fi
     if test -z "$CXX"; then CXX=cl; fi
     if test -z "$CPP"; then CPP="cl -E -nologo"; fi
     if test -z "$CXXCPP"; then CXXCPP="cl -TP -E -nologo"; ac_cv_prog_CXXCPP="$CXXCPP"; fi
@@ -191,18 +191,19 @@ case "$target" in
         x86_64)
             AS=ml64;
             ;;
         esac
     fi
     if test -z "$MIDL"; then MIDL=midl; fi
     ;;
 *-darwin*)
-    if test -z "$CC"; then CC=gcc-4.2; fi
-    if test -z "$CXX"; then CXX=g++-4.2; fi
+    # prefer gcc-4.2 to default cc on older xcode
+    MOZ_PATH_PROGS(CC, $CC gcc-4.2 gcc)
+    MOZ_PATH_PROGS(CXX, $CXX g++-4.2 g++)
     ;;
 esac
 fi
 
 COMPILE_ENVIRONMENT=1
 MOZ_ARG_DISABLE_BOOL(compile-environment,
 [  --disable-compile-environment
                           Disable compiler/library checks.],
